addons/binding/org.openhab.binding.fritzaha/src/main/java/org/openhab/binding/fritzaha/handler/FritzAhaBridgeHandler.java

```diff
@@ -0,0 +1,282 @@
+
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.fritzaha.FritzAhaBindingConstants;
+import org.openhab.binding.fritzaha.config.FritzAhaBridgeConfiguration;
+import org.openhab.binding.fritzaha.internal.discovery.FritzBoxDiscoveryService;
+import org.openhab.binding.fritzaha.internal.hardware.FritzahaWebInterface;
+import org.openhab.binding.fritzaha.internal.hardware.callbacks.FritzAhaDiscoveryCallback;
+import org.openhab.binding.fritzaha.internal.hardware.callbacks.FritzAhaUpdateXmlCallback;
+import org.openhab.binding.fritzaha.internal.model.DeviceModel;
+import org.openhab.binding.fritzaha.internal.model.SwitchModel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handler for a FRITZ!Box device. Handles polling of values from AHA devices.
+ * 
+ * @author Robert Bausdorf
+ * @since 2.0
+ */
+public class FritzAhaBridgeHandler extends BaseBridgeHandler {
+	/**
+	 * Logger
+	 */
+	private static Logger LOG = LoggerFactory
+			.getLogger(FritzAhaBridgeHandler.class);
+	/**
+	 * the refresh interval which is used to poll values from the fritzaha.
+	 * server (optional, defaults to 10 s)
+	 */
+	private long refreshInterval = 10;
+	/**
+	 * Interface object for querying the FRITZ!Box web interface
+	 */
+	private FritzahaWebInterface connection;
+	/**
+	 * Holder for last data received from the box.
+	 */
+	private Map<String, DeviceModel> deviceList;
+	/**
+	 * Job which will do the FRITZ!Box polling
+	 */
+	private FritzPolling pollingRunnable;
+	/**
+	 * Schedule for polling
+	 */
+	private ScheduledFuture<?> pollingJob;
+
+	/**
+	 * Constructor
+	 * 
+	 * @param bridge
+	 *            Bridge object representing a FRITZ!Box
+	 */
+	public FritzAhaBridgeHandler(Bridge bridge) {
+		super(bridge);
+		this.deviceList = new TreeMap<String, DeviceModel>();
+		this.pollingRunnable = new FritzPolling(this);
+	}
+
+	/**
+	 * Initializes the bridge.
+	 */
+	@Override
+	public void initialize() {
+		LOG.debug("About to initialize bridge "
+				+ FritzAhaBindingConstants.BRIDGE_FRITZAHA);
+		LOG.debug("lookup discovered boxes");
+		Bridge bridge = this.getThing();
+		FritzAhaBridgeConfiguration config = null;
+		if (bridge != null) {
+			String discoveredIp = bridge.getConfiguration().getProperties()
+					.get(FritzAhaBindingConstants.IP_ADDRESS).toString();
```

Comment:
No, what I meant is the value of the constant FritzAhaBindingConstants.IP_ADDRESS, which is ipAddress.
When a thing is created from the discovery result, all existing properties are automatically taken over (with the same key) - so I would expect that everything works as before if you simply remove this code here.